mlir/lib/Driver/CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+if(WIN32)
+    set(EXTERNAL_LIB Version)
+elseif(APPLE)
+    set(EXTERNAL_LIB z ncurses)
+elseif(UNIX)
+    set(EXTERNAL_LIB z tinfo)
+endif()
```

Comment:
We only need `z`. I deleted the rest.

```
FAILED: python_packages/quantum/mlir_quantum/_mlir_libs/libQuantumPythonCAPI.so.18git 
: && /usr/bin/clang++ -fPIC -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG  -Wl,-z,defs -Wl,-z,nodelete -fuse-ld=lld -Wl,--color-diagnostics   -Wl,-rpath-link,/home/erick.ochoalopez/Code/catalyst-latest/mlir/build/lib  -Wl,--gc-sections  -Xlinker -z -Xlinker defs -shared -Wl,-soname,libQuantumPythonCAPI.so.18git -o python_packages/quantum/mlir_quantum/_mlir_libs/libQuantumPythonCAPI.so.18git @CMakeFiles/QuantumPythonCAPI.rsp  && :
ld.lld: error: undefined symbol: crc32
>>> referenced by CRC.cpp
>>>               CRC.cpp.o:(llvm::crc32(unsigned int, llvm::ArrayRef<unsigned char>)) in archive /home/erick.ochoalopez/Code/catalyst-latest/mlir/llvm-project/build/lib/libLLVMSupport.a
>>> referenced by CRC.cpp
>>>               CRC.cpp.o:(llvm::crc32(llvm::ArrayRef<unsigned char>)) in archive /home/erick.ochoalopez/Code/catalyst-latest/mlir/llvm-project/build/lib/libLLVMSupport.a
>>> referenced by CRC.cpp
>>>               CRC.cpp.o:(llvm::JamCRC::update(llvm::ArrayRef<unsigned char>)) in archive /home/erick.ochoalopez/Code/catalyst-latest/mlir/llvm-project/build/lib/libLLVMSupport.a
```

Reply:
I'm curious how did these lines get here though? Is the `z` the missing dependency discovered by @josh146?

Reply:
No. `zlib` is different from `zstd`. @josh146 mentioned a dependencty to `zstd`. I suppose it could be either or. The LLVMSupport library uses them if compiled with `-DLLVM_ENABLE_ZLIB` and `-DLLVM_ENABLE_ZSTD`. These options may be set automatically through CMake or some other flag. Do you want me to try to remove the dependency on both `z` and `zstd`?

Reply:
Thanks! No need to change anything, I just wanted to know why they are there :)
Could you add a quick comment to the file?

Reply:
https://github.com/PennyLaneAI/catalyst/pull/216/commits/f99ece8c94a58a994fc07d4844fb4d6f1cbb1d49